packages/client/src/build-id-types.ts

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * Operations that can be passed to {@link WorkflowClient.updateWorkerBuildIdCompatibility}.
+ *
+ * @experimental
+ */
+export type BuildIdOperation =
+  | NewIdInNewDefaultSet
+  | NewCompatibleVersion
+  | PromoteSetByBuildId
+  | PromoteBuildIdWithinSet
+  | MergeSets;
+
+/**
+ * Adds a new Build Id into a new set, which will be used as the default set for
+ * the queue. This means all new workflows will start on this Build Id.
+ *
+ * @experimental
+ */
+interface NewIdInNewDefaultSet {
+  operation: 'newIdInNewDefaultSet';
+  buildId: string;
+}
+
+/**
+ * Adds a new Build Id into an existing compatible set. The newly added ID becomes
+ * the default for that compatible set, and thus new workflow tasks for workflows which have been
+ * executing on workers in that set will now start on this new Build Id.
+ *
+ * @experimental
+ */
+interface NewCompatibleVersion {
+  operation: 'newCompatibleVersion';
+  // The Build Id to add to an existing compatible set.
+  buildId: string;
+  // A Build Id which must already be defined on the task queue, and is used to
+  // find the compatible set to add the new id to.
+  existingCompatibleBuildId: string;
+  // If set to true, the targeted set will also be promoted to become the
+  // overall default set for the queue.
+  promoteSet?: boolean;
+}
+
+/**
+ * Promotes a set of compatible Build Ids to become the current
+ * default set for the task queue. Any Build Id in the set may be used to
+ * target it.
+ *
+ * @experimental
+ */
+interface PromoteSetByBuildId {
+  operation: 'promoteSetByBuildId';
+  buildId: string;
+}
+
+/**
+ * Promotes a Build Id within an existing set to become the default ID for that
+ * set.
+ *
+ * @experimental
+ */
+interface PromoteBuildIdWithinSet {
+  operation: 'promoteBuildIdWithinSet';
+  buildId: string;
+}
+
+/**
+ * Merges two sets into one set, thus declaring all the Build Ids in both as
+ * compatible with one another. The default of the primary set is maintained as
+ * the merged set's overall default.
+ *
+ * @experimental
+ */
+interface MergeSets {
+  operation: 'mergeSets';
+  // A Build Id which is used to find the primary set to be merged.
+  primaryBuildId: string;
+  // A Build Id which is used to find the secondary set to be merged.
+  secondaryBuildId: string;
+}
+
+/**
+ * Represents the sets of compatible Build Id versions associated with some
+ * Task Queue, as fetched by {@link WorkflowClient.getWorkerBuildIdCompatability}.
+ *
+ * @experimental
+ */
+export interface WorkerBuildIdVersionSets {
+  /**
+   * All version sets that were fetched for this task queue.
+   */
+  readonly versionSets: BuildIdVersionSet[];
+
+  /**
+   * Returns the default set of compatible Build Ids for the task queue these sets are
+   * associated with.
+   */
+  defaultSet(): BuildIdVersionSet;
+
+  /**
+   * Returns the overall default Build Id for the task queue these sets are
+   * associated with.
+   */
+  defaultBuildId(): string;
```

Comment:
Cool, I forgot getters exist. IMO it's... kinda wrong, since it hides arbitrary cost operations behind what looks like pure memory access, but, it's nice for interface compatibility.